src/lib/one-graph/netlify-graph-code-exporter-snippets.js

```diff
@@ -0,0 +1,543 @@
+
+  return fnName
+}
+
+const coercerFor = (type, name) => {
+  const typeName = print(type).replace(/\W+/gi, '').toLocaleLowerCase()
+
+  switch (typeName) {
+    case 'string':
+      return `${name}`
+    case 'int':
+      return `parseInt(${name})`
+    case 'float':
+      return `parseFloat(${name})`
+    case 'boolean':
+      return `${name} === 'true'`
+    default:
+      return `${name}`
+  }
+}
+
+const asyncFetcherInvocation = (operationDataList, pluckerStyle) => {
+  const invocations = operationDataList
+    .filter((operationData) => ['query', 'mutation', 'subscription'].includes(operationData.type))
+    .map((namedOperationData) => {
+      const params = (namedOperationData.operationDefinition.variableDefinitions || []).map(
+        (def) => def.variable.name.value,
+      )
+
+      const invocationParams = params.map((param) => `${param}: ${munge(param)}`)
+
+      const pluckers = {
+        get:
+          dotProp
+            .get(namedOperationData, 'operationDefinition.variableDefinitions', [])
+            .map((def) => {
+              const name = def.variable.name.value
+              const withCoercer = coercerFor(def.type, `event.queryStringParameters?.${name}`)
+              return `const ${munge(name)} = ${withCoercer};`
+            })
+            .join('\n  ') || '',
+        post:
+          dotProp
+            .get(namedOperationData, 'operationDefinition.variableDefinitions', [])
+            .map((def) => {
+              const name = def.variable.name.value
+              return `const ${munge(name)} = eventBodyJson?.${name};`
+            })
+            .join('\n  ') || '',
+      }
+
+      let variableValidation = ''
+
+      let requiredVariableCount = 0
+
+      if ((dotProp.get(namedOperationData, 'operationDefinition.variableDefinitions', []).length !== 0 || 0) > 0) {
+        const requiredVariableNames = namedOperationData.operationDefinition.variableDefinitions
+          .map((def) => (print(def.type).endsWith('!') ? def.variable.name.value : null))
+          .filter(Boolean)
+
+        requiredVariableCount = requiredVariableNames.length
+
+        // TODO: Filter nullable variables
+        const condition = requiredVariableNames
+          .map((name) => `${munge(name)} === undefined || ${munge(name)} === null`)
+          .join(' || ')
+
+        const message = requiredVariableNames.map((name) => `\`${name}\``).join(', ')
+
+        variableValidation = `  if (${condition}) {
+    return {
+      statusCode: 422,
+      body: JSON.stringify({
+        error: 'You must supply parameters for: ${message}'
+      }),
+    };
+  }`
+      }
+
+      return `${pluckerStyle === 'get' ? pluckers.get : pluckers.post}
+
+${requiredVariableCount > 0 ? variableValidation : ''}
+
+  const { errors: ${namedOperationData.name}Errors, data: ${namedOperationData.name}Data } =
+    await Netligraph.${operationFunctionName(namedOperationData)}({ ${invocationParams.join(', ')} }, accessToken);
```

Comment:
Netligraph

Reply:
Renamed this to be `NetlifyGraph`. Very clunky for the code in my opinion though.